Dockerfile

```diff
@@ -0,0 +1,85 @@
+# base node image
+FROM node:20-bookworm-slim as base
+
+# set for base and all layer that inherit from it
+ENV NODE_ENV production
+
+# Install openssl for Prisma
+RUN apt-get update && apt-get install -y fuse3 openssl sqlite3 ca-certificates
+
+# Install all node_modules, including dev dependencies
+FROM base as deps
+
+WORKDIR /myapp/heat-stack
+
+ADD heat-stack/package.json heat-stack/package-lock.json heat-stack/.npmrc ./
+RUN npm install --include=dev
+
+# Setup production node_modules
+FROM base as production-deps
+
+WORKDIR /myapp/heat-stack
+
+COPY --from=deps /myapp/heat-stack/node_modules /myapp/heat-stack/node_modules
+ADD heat-stack/package.json heat-stack/package-lock.json heat-stack/.npmrc ./
+RUN npm prune --omit=dev
+
+# Build the app
+FROM base as build
+
+WORKDIR /myapp/heat-stack
+
+COPY --from=deps /myapp/heat-stack/node_modules /myapp/heat-stack/node_modules
+
+ADD heat-stack/prisma .
+RUN npx prisma generate
+
+ADD heat-stack/. .
+RUN npm run build
+
+FROM python:3.12.3-slim-bookworm as rules
```

Comment:
`minor`: I wonder if this should be named `rules-engine-build` or similar and the existing `build` above should be `heat-stack-build` or similar?  Just to clarify that _both_ of these containers are for the build process and not the final image?

Reply:
I made the changes.  Yes, this is an intermediate build process.  heat-stack.yaml in github actions creates the final image.